src/components/atoms/ProgressBar.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { useEffect } from 'react';
+
+interface ProgressBarProps {
+  color?: string;
+  currentNumber: number;
+  maxNumber: number;
+  fullWidth?: string;
+  backgroundType?: 'image' | 'solid';
+  children?: React.ReactNode;
+}
+export const ProgressBar = ({
+  color,
+  currentNumber,
+  maxNumber,
+  fullWidth,
+  backgroundType = 'solid',
+  children,
+}: ProgressBarProps) => {
+  const [progress, setProgress] = React.useState((currentNumber / maxNumber || 0) * 100);
+  useEffect(() => {
+    if (progress > 100) {
+      setProgress(100);
+    }
+  }, [progress]);
```

Comment:
You don't need state for this.

```suggestion
  const progress = Math.min((currentNumber / maxNumber || 0) * 100, 100);
```